src/files.rs

```diff
@@ -0,0 +1,1044 @@
+use std::collections::HashMap;
+use std::fmt;
+use std::fs;
+use std::path::*;
+use std::sync::{Arc, Mutex};
+use std::thread;
+
+use crossbeam::channel::{unbounded, Receiver, Sender};
+use serde::{Deserialize, Serialize};
+use serde_json::Value;
+use tracing::debug;
+
+use crate::error::*;
+use crate::metrics::crap::crap;
+use crate::metrics::sifis::{sifis_plain, sifis_quantized};
+use crate::metrics::skunk::skunk_nosmells;
+use crate::output::*;
+use crate::utility::*;
+
+/// Struct containing all the metrics
+#[derive(Clone, Default, Debug, Serialize, Deserialize, Copy)]
+pub struct Metrics {
+    pub sifis_plain: f64,
+    pub sifis_quantized: f64,
+    pub crap: f64,
+    pub skunk: f64,
+    pub is_complex: bool,
+    pub coverage: f64,
+}
+
+impl Metrics {
+    pub fn new(
+        sifis_plain: f64,
+        sifis_quantized: f64,
+        crap: f64,
+        skunk: f64,
+        is_complex: bool,
+        coverage: f64,
+    ) -> Self {
+        Self {
+            sifis_plain,
+            sifis_quantized,
+            crap,
+            skunk,
+            is_complex,
+            coverage,
+        }
+    }
+}
+
+/// Struct with all the metrics computed for a single file
+#[derive(Clone, Default, Debug, Serialize, Deserialize)]
+#[allow(dead_code)]
+pub struct FileMetrics {
+    pub metrics: Metrics,
+    pub file: String,
+    pub file_path: String,
+}
+
+impl FileMetrics {
+    pub fn new(metrics: Metrics, file: String, file_path: String) -> Self {
+        Self {
+            metrics,
+            file,
+            file_path,
+        }
+    }
+
+    pub fn avg(m: Metrics) -> Self {
+        Self {
+            metrics: m,
+            file: "AVG".to_string(),
```

Comment:
Let's use `into` instead of `to_string`. Replace this everywhere else in the code